corehq/apps/enterprise/enterprise.py

```diff
@@ -635,3 +634,148 @@ def rows_for_domain(self, domain_obj):
+                    app_name_by_id[app_id] = Application.get_db().get(app_id).get('name')
+                rows.append([
+                    user['username'],
+                    domain,
+                    app_name_by_id[app_id],
+                    latest_version,
+                    build['build_version'],
+                    self.format_date(DateTimeProperty.deserialize(build['build_version_date'])),
+                ])
+
+        return rows
+
+    def total_for_domain(self, domain):
+        apps = get_app_ids_in_domain(domain)
+        total_last_builds = 0
+        total_out_of_date = 0
+
+        for user, build, latest_version in self._get_user_builds(domain, apps):
+            total_last_builds += 1
+            if is_out_of_date(str(build['build_version']), str(latest_version)):
+                total_out_of_date += 1
+
+        return total_out_of_date, total_last_builds
+
+    def _get_user_builds(self, domain, apps):
+        user_query = (UserES()
+            .domain(domain)
+            .mobile_users()
+            .source([
+                'username',
+                'reporting_metadata.last_builds',
+            ]))
+        for user in user_query.run().hits:
+            last_builds = user.get('reporting_metadata', {}).get('last_builds', [])
+            for build in last_builds:
+                app_id = build.get('app_id')
+                build_version = build.get('build_version')
+                if app_id not in apps or not build_version:
+                    continue
+                build_version_date = DateTimeProperty.deserialize(build.get('build_version_date'))
+                all_builds = self.get_app_builds(domain, app_id, limit=self.INITIAL_QUERY_LIMIT)
+                latest_version = self.get_latest_build_version_at_time(all_builds, build_version_date)
+                if not latest_version:
+                    all_builds = self.get_app_builds(domain, app_id)
+                    latest_version = self.get_latest_build_version_at_time(all_builds, build_version_date)
+                yield user, build, latest_version
+
+    def get_app_builds(self, domain, app_id, limit=None):
+        if app_id not in self.builds_by_app_id:
+            app_es = (
+                AppES()
+                .domain(domain)
+                .is_build()
+                .app_id(app_id)
+                .sort('version', desc=True)
+                .is_released()
+                .source(['_id', 'version', 'last_released', 'built_on'])
+            )
+            if limit:
+                app_es = app_es.size(limit)
+            self.builds_by_app_id[app_id] = app_es.run().hits
+
+        return self.builds_by_app_id[app_id]
+
+    def get_latest_build_version_at_time(self, all_builds, time):
+        """
+        Get the latest build version at the time
+
+        :param all_builds: List of raw build documents sorted by version in descending order
+        :param time: The date of the build version to compare against
```

Comment:
Might be worth specifying what date format is expected here. Seems like it's whatever is returned by `DateTimeProperty.deserialize`.